L12-Gas-optimizations.md

```diff
@@ -125,38 +124,81 @@ Libraries are stateless contracts that don't store any state. Now when you call
 There is a small caveat however. If you are writing your own libraries, you will need to deploy them and pay gas - but once deployed, it can be reused by other smart contracts without deploying it themselves. Since they don't store any state, libraries only need to be deployed once to the blockchain and are assigned an address that the Solidity compiler is smart enough to figure out itself. Therefore, if you use libraries from OpenZeppelin for example, they will not add to your deployment cost.
 
 ## Short Circuiting Conditionals
-
 If you are using (||) or (&&) it's better to write your conditions in such a way so that the least functions/variable values are executed or retrieved in order to determine if the entire statement is true or false.
 
 Since conditional checks will stop the second they find the first value which satisfies the condition, you should put the variables most likely to validate/invalidate the condition first. In OR conditions (||), try to put the variable with the highest likelihood of being `true` first, and in AND conditions (&&), try to put the variable with the highest likelihood of being `false` first. As soon as that variable is checked, the conditional can exit without needing to check the other values, thereby saving gas.
 
 <Quiz questionId="dc5793a3-18c1-41f7-987d-cc3ae3bc0f18" />
 
 ## Free up Storage
-
 Since storage space costs gas, you can actually free up storage and delete unnecessary data to get gas refunds. So if you no longer need some state values, use the `delete` keyword in Solidity for some gas refunds.
 
 ## Short Error Strings
-
 Make sure that the error strings in your require statements are of very short length, the more the length of the string, the more gas it will cost.
 
 ```solidity
 require(counter >= 100, "NOT REACHED"); // Good
 require(balance >= amount, "Counter is still to reach the value greater than or equal to 100, ............................................";
 ```
-
-The first requirement is more gas optimized than the second one.
+The first requirement is more gas optimized than the second one
 
 > NOTE: In newer versions of Solidity, there are now custom errors using the `error` keyword which behave very similar to `events` and can achieve similar gas optimizations.
+----
+
+
+## Declaring the constructor payable
+Declaring the constructor as payable can potentially save a small amount of gas when deploying a contract in Solidity. This is because the Solidity compiler will skip a set of opcodes that check whether the constructor is payable at runtime if the constructor is declared as payable, resulting in a smaller contract bytecode and fewer gas units required to execute it.What this actually does is that it skips this set of opcodes :
+* ```CALLVALUE```
```

Comment:
I appreciate the effort for this. I think it would be really cool if we dived into a bit of Yul in the Advanced Section or made another for Yul.

Reply:
> I appreciate the effort for this. I think it would be really cool if we dived into a bit of Yul in the Advanced Section or made another for Yul.

yeah, we should do this, how should we do this

Reply:
Maybe leave in small content for people to be interested in this and make a sub-course for Yul? 

Since Yul is almost like a language on its own, it deserves its own space, but making the content for Yul will be difficult.

Reply:
lets take this discussion to discord